pkg/cmd/audit/io.go

```diff
@@ -405,3 +407,42 @@ func PrintSummary(w io.Writer, events []*auditv1.Event) {
 	fmt.Fprintf(w, "count: %d, first: %s, last: %s, duration: %s\n", len(events),
 		first.RequestReceivedTimestamp.Time.Format(time.RFC3339), last.RequestReceivedTimestamp.Time.Format(time.RFC3339), duration.String())
 }
+
+// IsEquivalentAuditURI is fuzzy matcher that allows equivalence on non-exact matches.  This is important for watches and
+// for lists since they can pass a resourceversion and timeout which always diffs, but is rarely importantly different
+func IsEquivalentAuditURI(lhs, rhs string) bool {
+	if lhs == rhs {
+		return true
+	}
+	lhsURL, err := url.Parse("https://example.com" + lhs)
+	if err != nil {
+		panic(err)
+	}
+	rhsURL, err := url.Parse("https://example.com" + rhs)
+	if err != nil {
+		panic(err)
+	}
+	if lhsURL.Path != rhsURL.Path {
+		return false
+	}
+	lhsQueries := lhsURL.Query()
+	rhsQueries := lhsURL.Query()
+
+	lhsKeys := sets.StringKeySet(lhsQueries)
+	rhsKeys := sets.StringKeySet(rhsQueries)
+	lhsKeys.Delete("timeout", "timeoutSeconds", "resourceVersion")
+	rhsKeys.Delete("timeout", "timeoutSeconds", "resourceVersion")
```

Comment:
how about `limit` and `continue`,  do we need to delete it as well? On the other hand, what if we simply compare the request path, is the count going to be wrong?

Reply:
or do we want to differentiate by some query parameters?

Reply:
limit we don't need to remove since it should be consistent.  continue we can probably remove since the token varies.

We need to compare some query parameters since they are used to pass filtering options and the distinctions there matter.  I think mattering by default makes sense.